src/SpatialNavigation.ts

```diff
@@ -1171,7 +1190,11 @@ class SpatialNavigationService {
     const componentToRemove = this.focusableComponents[focusKey];
 
     if (componentToRemove) {
-      const { parentFocusKey } = componentToRemove;
+      const { parentFocusKey, onUpdateFocus } = componentToRemove;
+
+      onUpdateFocus(false);
```

Comment:
This helps to set proper state on the component when it is removed. Without this, it gets stuck with `focused: true` during the "double mount" of React v18 StrictMode.